src/commands/update.js

```diff
@@ -0,0 +1,203 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const { Command, flags } = require('@oclif/command')
+const inquirer = require('inquirer')
+const { cli } = require('cli-ux')
+const ora = require('ora')
+const { prompt, getNpmLatestVersion, getNpmLocalVersion } = require('../helpers')
+
+class UpdateCommand extends Command {
+  /**
+   * List the plugins that have updates.
+   *
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   */
+  async __list (plugins) {
+    const columns = {
+      plugin: {
+        width: 10,
+        get: row => `${row.name}`
+      },
+      current: {
+        minWidth: 10,
+        get: row => `${row.currentVersion}`
+      },
+      latest: {
+        get: row => `${row.latestVersion}`
+      }
+    }
+
+    cli.table(plugins, columns)
+  }
+
+  /**
+   * Install the plugins that have updates.
+   *
+   * @private
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   * @param {boolean} needsConfirm true to show confirmation prompt
+   */
+  async __install (plugins, needsConfirm) {
+    await this.__list(plugins)
+    let _doUpdate = true
+
+    this.log() // newline
+
+    if (needsConfirm) {
+      _doUpdate = await prompt(`Update ${plugins.length} plugin(s)?`)
+    }
+
+    if (_doUpdate) {
+      // install the plugins in sequence
+      for (const plugin of plugins) {
+        await this.config.runCommand('plugins:install', [plugin.name])
+      }
+    }
+  }
+
+  /**
+   * Install the plugins that have  updates, with an interactive install.
+   *
+   * @private
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   */
+  async __interactiveInstall (plugins) {
+    const inqChoices = plugins
+      .map(plugin => { // map to expected inquirer format
+        return {
+          name: `${plugin.name}@${plugin.latestVersion}`,
+          value: plugin.name
+        }
+      })
+
+    const response = await inquirer.prompt([{
+      name: 'plugins',
+      message: 'Select plugins to update',
+      type: 'checkbox',
+      choices: inqChoices
+    }])
+
+    // install the plugins in sequence
+    for (const plugin of response.plugins) {
+      await this.config.runCommand('plugins:install', [plugin])
+    }
+  }
+
+  /**
+   * Process the plugins, determine if they need updates or warnings.
+   *
+   * @param {object} cliRoot the root folder of the cli
+   * @param {Array<string>} corePlugins package names of the core plugins
+   * @param {Array<InstalledPlugin>} installedPlugins
+   * @returns {Array<ToUpdatePlugin}
+   */
+  async __processPlugins () {
```

Comment:
nit: At some point some of us started putting double __methodPrefix to signify that something is private or not exported.  Personally, I don't really like this practice.
https://github.com/airbnb/javascript#naming--leading-underscore

Reply:
Initially I wanted the functions into a the helper module. But since these functions heavily use base command functionality (this.config, etc), I kept it here. Until we have private functions in EcmaScript classes, this visual convention would have to do. The only public contract we have with devs is the run() function, nothing else.